checkov/terraform/checks/resource/azure/NSGRulePortAccessRestricted.py

```diff
@@ -26,6 +26,9 @@ def is_port_in_range(self, conf):
         return False
 
     def scan_resource_conf(self, conf):
+        if any("dynamic" in s for s in conf):
```

Comment:
This test is really wide. Why not just `if "dynamic" in conf`?

Currently dynamic can only be used for `security_rule` so this will not skip other cases. What do you think about an additional check that dynamic is used for `security_rule`?

Reply:
Hey @davenicoll !

Thanks for the contribution. However, this change will mean if the resource has **any** dynamic blocks in it, it will return as `UNKNOWN`. IMO it would be better to check the `rule_conf` - verify it is a dict before checking fields inside `rule_conf`. WDYT?